File: Lab4/Question1/src/Student.java

public class Student {
	private int ID;
	private boolean sleeping;

	public Student(int ID) {// Initialize student with ID
		this.ID = ID;
		sleeping = false;
	}

	public int getID() {// Get the ID of student
		return ID;
	}

	public boolean isSleeping() {// Check if the student is sleeping
		return sleeping;
	}

	public void fallAsleep() {// Make the student fall asleep
		if (!isSleeping())
			sleeping = true;
	}

	public void wakeUp() {// Make the student wake up
		if (isSleeping())
			sleeping = false;
	}

	public static void testStudent() {
		Student s = new Student(1234567890);
		//Testing get information functions
		System.out.println(s.getID() == 1234567890);
		System.out.println(s.isSleeping() == false);
		//Testing fall asleep function
		s.fallAsleep();
		System.out.println(s.isSleeping() == true);
		s.fallAsleep();
		System.out.println(s.isSleeping() == true);
		//Testing wakeUp function
		s.wakeUp();
		System.out.println(s.isSleeping() == false);
		s.wakeUp();
		System.out.println(s.isSleeping() == false);
	}
}

File: Lab4/Student/Student.kt
class Student {
    private val ID: Int
    private var sleeping: Boolean

    constructor(ID: Int) {
        this.ID = ID
        sleeping = false
    }

    fun getID(): Int {
        return ID
    }

    fun isSleeping(): Boolean {
        return sleeping
    }

    fun fallAsleep() {
        if (!isSleeping())
            sleeping = true
    }

    fun wakeUp() {
        if (isSleeping())
            sleeping = false
    }

    companion object {
        fun testStudent() {
            val stu = Student(1234567890)
            println(stu.getID() == 1234567890)
            println(stu.isSleeping() == false)
            stu.fallAsleep()
            println(stu.isSleeping() == true)
            stu.fallAsleep()
            println(stu.isSleeping() == true)
            stu.wakeUp()
            println(stu.isSleeping() == false)
            stu.wakeUp()
            println(stu.isSleeping() == false)
        }
    }
}